bucket_map/src/bucket_storage.rs

```diff
@@ -262,6 +263,37 @@ impl BucketStorage {
         }
     }
 
+    fn get_mmap_count() -> Option<usize> {
+        let pid = std::process::id();
+        let map_path = format!("/proc/{}/maps", pid);
+        let output = std::process::Command::new("wc")
```

Comment:
we can avoid shelling out with something like https://doc.rust-lang.org/rust-by-example/std_misc/file/read_lines.html.  just toss a `.count()` on after `.lines()`

Reply:
we _may_ want to copy the maps file to a temporary first, to avoid being bitten by non-atomic reads in procfs

Reply:
It looks like readlines is slower than `wc -l`.

```
Time elapsed is: 3.10014196s
```
I guess /proc/pid/maps are in memory kernel data structure, which is faster to read than the file on disk.

Reply:
> I guess /proc/pid/maps are in memory kernel data structure, which is faster to read than the file on disk.

yeah it is.  also means it can be updated from beneath you.  which is where the non-atomicity issues come in that i keep crowing on about :sweat_smile:

Reply:
Yes. But since this is used to report a metric, a small difference, caused by not being atomic, would be fine, I think.